device/ospf/ospfv2/interface/interface.yaml

```diff
@@ -0,0 +1,225 @@
+components:
+  schemas:
+    Ospfv2.Interface:
+      description: >-
+        Configuration for single OSPFv2 interface.
+      type: object
+      required: [ipv4_name, name]
+      properties:
+        name:
+          x-include: ../common/common.yaml#/components/schemas/Named.Object/properties/name
+          x-field-uid: 1
+        ipv4_name:
+          description: >-
+            The globally unique name of the IPv4 interface connected to the DUT.
+            This name must match the "name" field of the "ipv4_addresses" on top which this OSPFv2 interface is configured.          
+          type: string
+          x-constraint:
+          - '/components/schemas/Device.Ipv4/properties/name'
+          x-field-uid: 2
+        choice:
+          description: |-
+            The OSPF Area ID identifies the routing area to which the host belongs. Area ID type can be following format.
+            - area_id: A 32-bit number identifying the area.
+            - area_id_as_ip: The Area ID in IPv4 address format.
+          type: string
+          default: area_id
+          x-field-uid: 3
+          x-enum:
+            area_id:
+              x-field-uid: 1
+            area_id_as_ip:
+              x-field-uid: 2
+        area_id:
+          description: >-
+            The Area ID.
+          type: integer
+          format: uint32
+          x-field-uid: 4
+        area_id_as_ip:
+          description: >-
+            The Area ID in IPv4 address format.
+          type: string
+          format: ipv4
+          x-field-uid: 5
+        network_type:
+          description: |- 
+            The OSPF network link type options.
+            - Point to Point: 
+            - Broadcast: 
+            - Point to Multipoint: In this case, at least a neigbor to be configured. 
```

Comment:
I am thinking whether we should keep out the p2mp option for now. I dont think it is much used and might be difficult to implement and test. Not sure if IxNetwork mode of one neighbor per p2mp is very sensible ( I see you do have as array which feels right  )and might be good to avoid that completely,